pkg/harvester-manager/models/harvesterhci.io.management.cluster.js

```diff
@@ -29,199 +41,47 @@ export default class HciCluster extends ProvCluster {
+      return stateDisplay(STATES_ENUM.DENIED);
     }
 
-    throw new Error(`Unsupported value for ${ SETTING.UI_OFFLINE_PREFERRED }: 'uiOfflinePreferred'`);
+    return stateDisplay(this.state);
   }
 
-  /**
-   * Determine the harvester plugin's package name and url for clusters that provide the plugin
-   */
-  _supportedClusterPkgDetails(uiInfo, clusterId) {
-    let pkgName = `${ HARVESTER_NAME }-${ uiInfo['ui-plugin-bundled-version'] }`;
-    const fileName = `${ pkgName }.umd.min.js`;
-    let pkgUrl;
-
-    if (uiInfo['ui-source'] === 'bundled' ) { // offline bundled
-      pkgUrl = `/k8s/clusters/${ clusterId }/v1/harvester/plugin-assets/${ fileName }`;
-    } else if (uiInfo['ui-source'] === 'external') {
-      if (uiInfo['ui-plugin-index']) {
-        pkgUrl = uiInfo['ui-plugin-index'];
-
-        // When using an external address, the pkgName should also be get from the url
-        const names = pkgUrl.split('/');
-        const jsName = names[names.length - 1];
-
-        pkgName = jsName?.split('.umd.min.js')[0];
-      } else {
-        throw new Error('Harvester cluster requested the plugin at `ui-plugin-index` is used, however did not provide a value for it');
+  async goToCluster() {
+    this.currentRouter().push({
+      name:   `${ VIRTUAL }-c-cluster-resource`,
+      params: {
+        cluster:  this.status.clusterName,
+        product:  VIRTUAL,
+        resource: HCI.DASHBOARD // Go directly to dashboard to avoid blip of components on screen
       }
-    }
-
-    return {
-      pkgUrl,
-      pkgName
-    };
+    });
   }
 
-  _overridePkgDetails() {
-    // Support loading the pkg from a locally, or other, address
-    // This helps testing of the harvester plugin when packaged up, instead of directly imported
-    const harvesterPkgUrl = process.env.harvesterPkgUrl;
-
-    if (!harvesterPkgUrl) {
+  async setSupportedHarvesterVersion() {
+    if (this._isSupportedHarvester !== undefined) {
       return;
     }
-    const parts = harvesterPkgUrl.replace('.umd.min.js', '').split('/');
-    const pkgNameFromUrl = parts.length > 1 ? parts[parts.length - 1] : null;
-
-    if (pkgNameFromUrl) {
-      return {
-        pkgUrl:  harvesterPkgUrl,
-        pkgName: pkgNameFromUrl
-      };
-    }
-  }
-
-  async _pkgDetails() {
-    const overridePkgDetails = this._overridePkgDetails();
-
-    if (overridePkgDetails) {
-      return overridePkgDetails;
-    }
-
-    const clusterId = this.mgmt.id;
-    const uiInfo = await this._getUiInfo(clusterId);
 
-    return uiInfo ? this._supportedClusterPkgDetails(uiInfo, clusterId) : this._legacyClusterPkgDetails();
-  }
+    const url = `/k8s/clusters/${ this.status.clusterName }/v1`;
 
-  async loadClusterPlugin() {
-    // Skip loading if it's built in
-    const plugins = this.$rootState.$plugin.getPlugins();
-    const loadedPkgs = Object.keys(plugins);
+    try {
+      const setting = await this.$dispatch('request', { url: `${ url }/${ HCI.SETTING }s/server-version` });
```

Comment:
Not an action point, but the lingering `s` because of that singular to plural change for requests is nerve-wrecking 🤣

Reply:
Ah this is part of the blocking PR, will change later.